GCRCatalogs/instance_catalog.py

```diff
@@ -246,47 +265,82 @@ def _generate_native_quantity_list(self):
         native_quantities.append('gal/total_id')
         return native_quantities
 
-
-    def _get_data(self, obj_type):
+    def _pd_read_table(self, obj_type, **kwargs):
+        return pd.read_table(
+            self._object_files[obj_type],
+            delim_whitespace=True,
+            names=[c[0] for c in self._col_names[obj_type]],
+            dtype=dict(self._col_names[obj_type]),
+            **kwargs
+        )
+
+    def _load_legacy_gal_catalog(self, obj_type):
+        if '_legacy_gal_line_index' not in self._data:
+            path = self._object_files[obj_type]
+            this_open = gzip.open if path.endswith('.gz') else open
+            with this_open(path, 'rb') as f:
+                for index, line in enumerate(f):
+                    if b' agnSED/' in line:
+                        self._data['_legacy_gal_line_index'] = index
```

Comment:
Just checking: are we absolutely certain that the AGN are all piled up at the end of the catalog?  I am specifically worried about the sprinkler breaking this assumption.

Reply:
This is at least true for all the Run 1.2 instance catalogs, but I am not sure if this assumption is *always* true. However, given that (1) this assumption does help a lot in term of load performance though, (2) if there are galaxy entries appearing after an AGN line, the reader will throw an error rather than proceed silently, and (3) the instance catalog will be in the new format for Run 2.0 (is that right?), I think it is ok to leave the legacy support as it. What do you think, @danielsf?

Reply:
If it's true for Run 1.2, then I agree it is safe to use.  We should not be going back to the legacy format any time soon (or, really, at all).